src/index.ts

```diff
@@ -44,18 +45,21 @@ function createRdfEmitter(program: Program) {
     emit,
   };
 
-  function emit() {
-    navigateProgram(program, {
+  function emit() 
+  {
+    // CLASS DEF 
+    navigateProgram(program, 

```

Comment:
This approach of navigating the program 3 times has a rather significant drawback I think - traversal is probably the most costly operation we're doing in this emitter, and we're traversing 3 times. Doing everything in one traversal would be more performant, but also probably more clear. You could accomplish this by creating three arrays to start - shapes, properties, and constraints (or whatever you call those sections). Then, instead of calling addQuad directly, push the arguments to the appropriate array. Then once at the end, iterate over these arrays and call addQuad.

While performance probably isn't much of a concern given the size of the cadl programs we're talking about here, it would be good to be efficient when we can.